commcare_connect/opportunity/views.py

```diff
@@ -1376,3 +1363,308 @@ def sync_deliver_units(request, org_slug, opp_id):
+        },
+    )
+    return response
+
+
+def get_user_visit_counts(opportunity_access_id: int, date):
+    opportunity_access = OpportunityAccess.objects.get(id=opportunity_access_id)
+    visit_count_kwargs = dict(approved=Count("id", filter=Q(status=VisitValidationStatus.approved)))
+    if opportunity_access.opportunity.managed:
+        visit_count_kwargs = dict(
+            pending_review=Count(
+                "id",
+                filter=Q(
+                    status=VisitValidationStatus.approved,
+                    review_status=VisitReviewStatus.pending,
+                ),
+            ),
+            disagree=Count(
+                "id",
+                filter=Q(
+                    status=VisitValidationStatus.approved,
+                    review_status=VisitReviewStatus.disagree,
+                ),
+            ),
+            approved=Count(
+                "id",
+                filter=Q(
+                    status=VisitValidationStatus.approved,
+                    review_status=VisitReviewStatus.agree,
+                ),
+            ),
+        )
+
+    filter_kwargs = {"opportunity_access": opportunity_access}
+    if date:
+        filter_kwargs.update({"visit_date__date": date})
+
+    user_visit_counts = UserVisit.objects.filter(**filter_kwargs).aggregate(
+        **visit_count_kwargs,
+        pending=Count("id", filter=Q(status=VisitValidationStatus.pending)),
+        rejected=Count("id", filter=Q(status=VisitValidationStatus.rejected)),
+        flagged=Count(
+            "id", filter=Q(flagged=True, status__in=[VisitValidationStatus.approved, VisitValidationStatus.rejected])
+        ),
+        total=Count("*"),
+    )
+    return user_visit_counts
+
+
+class VisitVerificationTableView(OrganizationUserMixin, SingleTableView):
+    model = UserVisit
+    table_class = UserVisitVerificationTable
+    template_name = "opportunity/user_visit_verification_table.html"
+    exclude_columns = []
+
+    def get_paginate_by(self, table_data):
+        return self.request.GET.get("per_page", 10)
+
+    def get_table(self, **kwargs):
+        kwargs["exclude"] = self.exclude_columns
+        self.table = super().get_table(**kwargs)
+        return self.table
+
+    def dispatch(self, request, *args, **kwargs):
+        response = super().dispatch(request, *args, **kwargs)
+        url = reverse(
+            "opportunity:user_visits_list",
+            args=[request.org.slug, self.kwargs["opp_id"], self.kwargs["pk"]],
+        )
+        query_params = request.GET.urlencode()
+        response["HX-Replace-Url"] = f"{url}?{query_params}"
+        return response
+
+    def get_table_kwargs(self):
+        kwargs = super().get_table_kwargs()
+        kwargs["organization"] = self.request.org
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+        user_visit_counts = get_user_visit_counts(self.kwargs["pk"], self.filter_date)
+        tabs = [
```

Comment:
if this is going to be completely overwritten for program manager, it might be clearer in an `else` block

Reply:
Fixed. [Refactor tabs for nm in else block](https://github.com/dimagi/commcare-connect/pull/571/commits/54bf4eb01e2d89cd54f89770468ac4312715ee25)